frontend-ssr/pages/submissions.tsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+import Head from 'next/head';
+import Footer from '~/components/Footer';
+import Header from '~/components/Header';
+
+import ExerciseSubmission from '../components/exercise-submissions/ExerciseSubmission';
+import ExerciseService from '../services/ExerciseSubmissions.service';
+import {
+  loadSubmissions,
+  searchSubmissions,
+} from '~/redux/exercise-submissions/exercise-submissions.actions';
+import ExerciseSubmissionSkeleton from '~/components/exercise-submissions/ExerciseSubmissionSkeleton';
+import SweetAlertService from '../services/sweet-alert/SweetAlert.service';
+import Search from '~/components/Search';
+import { RootState } from '../redux/root.reducer';
+
+import styles from './submissions.module.scss';
+
+interface State {
+  loading: boolean;
+  prevY: number;
+}
+
+class Submissions extends React.Component<ConnectedProps<typeof connector>, State> {
+  private observer: IntersectionObserver;
+
+  private buttonRef = React.createRef<HTMLButtonElement>();
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: props.submissions.submissions === undefined,
+      prevY: 0,
+    };
+  }
+
+  componentDidMount() {
+    let { submissions } = this.props;
+
+    if (submissions.submissions === undefined) {
+      this.getData();
+    }
+
+    const options = {
+      threshold: 0.7,
+    };
+
+    this.observer = new IntersectionObserver(this.loadMore, options);
+    this.observeNextPage();
+  }
+
+  componentWillUnmount() {
+    this.observer.disconnect();
```

Comment:
Kudos for disconnecting when we leave.

Reply:
However, we should also disconnect when we reach the end: meaning we have no data left on the server.

For this we should add a new property in store, which tells us if we're at the end or not. This way, if we navigate back-forward to this page, it will know we're still at the end and **should not** initialize the Observer.